src/v/kafka/server/group_manager.cc

```diff
@@ -95,6 +104,31 @@ ss::future<> group_manager::stop() {
     return _gate.close();
 }
 
+void group_manager::detach_partition(const model::ntp& ntp) {
+    klog.debug("detaching group metadata partition {}", ntp);
+    std::vector<group_ptr> groups;
+    groups.reserve(_groups.size());
+    for (auto& group : _groups) {
+        groups.push_back(group.second);
+    }
+
+    for (auto& gr : groups) {
+        // skip if group is not managed by current NTP
+        if (gr->partition()->ntp() != ntp) {
+            continue;
+        }
+        auto it = _groups.find(gr->id());
+        if (it == _groups.end()) {
+            continue;
+        }
+
+        vlog(klog.trace, "Removed group {}", gr);
+        _groups.erase(it);
```

Comment:
what's not clear to me is if removing a group here races with other operations that create and register groups.

Reply:
I've audited all the code paths where group would be recreated if not available. All of  the continuations have `validate_group_status` function call. Even if group would be removed and then the continuation would be scheduled after that it will not be recreated since `validate_group_status` will return an error. All fibers that use a group which was deleted will return an error for group operations. Consumer will have to reach out to new coridantor.